.github/workflows/publish.yml

```diff
@@ -0,0 +1,54 @@
+---
+name: Release
+
+on:
+  release:
+    types:
+      - published
+
+concurrency:
+  group: release-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  hackage:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Haskell
+        id: setup-haskell-cabal
+        uses: haskell/actions/setup@v2
+        with:
+          # I faced the following issue with GHC-9.2 and and I managed to solved
+          # it by going back to GHC-8.10.7 just to generate the docs:
+          # - https://github.com/haskell/cabal/issues/8326
+          ghc-version: "8.10.7"
+          cabal-version: "3.6"
+
+      - name: Generate sdist
+        run: cabal sdist
+
+      - name: Upload candidate package to Hackage
+        run: |
+          SDIST_PATH=$(ls dist-newstyle/sdist/dotenv-*.tar.gz)
+          curl -X POST \
+           -H 'Accept: text/plain' \
+           -H 'Authorization: X-ApiKey ${{ secrets.HACKAGE_API_KEY }}' \
+           --form "package=@${SDIST_PATH}" \
+           "https://hackage.haskell.org/packages/candidates"
+
+      - name: Generate documentation
+        run: cabal haddock --haddock-for-hackage --enable-documentation
+
+      - name: Upload Docs for candidateto Hackage
+        run: |
+          TAG_VERSION=${{ github.ref_name }}
+          VERSION=${TAG_VERSION:1}
+          DOCS_PATH=$(ls dist-newstyle/dotenv-*-docs.tar.gz)
+          curl -X PUT \
+           -H 'Content-Type: application/x-tar' \
+           -H 'Content-Encoding: gzip' \
+           -H 'Authorization: X-ApiKey ${{ secrets.HACKAGE_API_KEY }}' \
```

Comment:
@CristhianMotoche based on the official documentation, GH automatically redacts secrets:

> Warning: GitHub automatically redacts secrets printed to the log, but you should avoid printing secrets to the log intentionally.

However, most of the examples found in the documentation define an environment variable based on a secret and then reference the variable in the script, here is a code snippet from GH:

```
steps:
  - shell: bash
    env:
      SUPER_SECRET: ${{ secrets.SuperSecret }}
    run: |
      example-command "$SUPER_SECRET"
```

Therefore, I would recommend following the same approach here. What do you think?

https://docs.github.com/en/actions/security-guides/encrypted-secrets#example-using-bash

Reply:
Good catch @sestrella Thanks for the tip!

Reply:
Hey @sestrella Regardless the GH documentation, I see the secrets are actually hidden somehow. Take a look at this sample: https://github.com/stackbuilders/dotenv-hs/actions/runs/3735107653/jobs/6337959409#step:7:10 I'm going to implement  your suggestion anyway, though. To avoid any possible vulnerability.

Reply:
Right, GH automatically redacts secrets, however, sensitive data is usually passed as environment variables to bash scripts, so I think that is why the documentation recommends that approach. Thank you for taking care of this!